zebra/zebra_rib.c

```diff
@@ -3083,16 +3083,15 @@ void rib_delete(afi_t afi, safi_t safi, vrf_id_t vrf_id, int type,
 	/* Lookup route node. */
 	rn = srcdest_rnode_lookup(table, p, src_p);
 	if (!rn) {
-		char src_buf[PREFIX_STRLEN];
-
-		if (src_p && src_p->prefixlen)
-			prefix2str(src_p, src_buf, sizeof(src_buf));
-		else
-			src_buf[0] = '\0';
-
 		if (IS_ZEBRA_DEBUG_RIB) {
+			char src_buf[PREFIX_STRLEN];
 			struct vrf *vrf = vrf_lookup_by_id(vrf_id);
 
+			if (src_p && src_p->prefixlen)
+				prefix2str(src_p, src_buf, sizeof(src_buf));
+			else
+				src_buf[0] = '\0';
+
 			zlog_debug("%s[%d]:%pFX%s%s doesn't exist in rib",
```

Comment:
Why not just use `%pFX`?

Example:

```c
if (src_p && src_p->prefixlen)
  zlog_debug("from %pFX", src_p);
else
  zlog_debug("...");
```

Reply:
I didn't use it because it would require a bit more rework and prefix2str was good enough for the moment I thought, as that I was trying to just limit work done for debug to debug if statement